lib/src/Collection.cpp

```diff
@@ -247,11 +247,24 @@ pplx::task<shared_ptr<Document>> Collection::CreateDocumentAsync(
 	{
 		value json_response = response.extract_json().get();
 
-		if (response.status_code() == status_codes::Created)
-		{
-			return DocumentFromJson(json_response);
-		}
+		for (size_t i = 0; i < MAX_QUERY_RETRY_CNT; ++i) {

```

Comment:
can you move FOR loop, such that call to http_client.request is inside it? Currently, we duplicate code for request() in two places (original place which puts you in async code, and then you iterate inside). This is rather interesting problem (chaining async calls in loop), maybe we should steal do_while from [here](https://www.codeproject.com/articles/585998/csharp-style-async-await-in-cpl) ? Consider that